cmd/monaco/download/download_configs.go

```diff
@@ -274,18 +274,24 @@ type downloadFn struct {
 	classicDownload      func(context.Context, client.ConfigClient, string, api.APIs, classic.ContentFilters) (project.ConfigsPerType, error)
 	settingsDownload     func(context.Context, client.SettingsClient, string, settings.Filters, ...config.SettingsType) (project.ConfigsPerType, error)
 	automationDownload   func(context.Context, client.AutomationClient, string, ...config.AutomationType) (project.ConfigsPerType, error)
-	bucketDownload       func(context.Context, client.BucketClient, string) (project.ConfigsPerType, error)
+	bucketDownload       func(client.BucketClient) Downloadable
 	documentDownload     func(context.Context, client.DocumentClient, string) (project.ConfigsPerType, error)
 	openPipelineDownload func(context.Context, client.OpenPipelineClient, string) (project.ConfigsPerType, error)
 	segmentDownload      func(context.Context, segment.DownloadSegmentClient, string) (project.ConfigsPerType, error)
 	sloDownload          func(context.Context, slo.DownloadSloClient, string) (project.ConfigsPerType, error)
 }
 
+type Downloadable interface {
+	Download(ctx context.Context, projectName string) (project.ConfigsPerType, error)
+}
+
 var defaultDownloadFn = downloadFn{
-	classicDownload:      classic.Download,
-	settingsDownload:     settings.Download,
-	automationDownload:   automation.Download,
-	bucketDownload:       bucket.Download,
+	classicDownload:    classic.Download,
+	settingsDownload:   settings.Download,
+	automationDownload: automation.Download,
+	bucketDownload: func(bucketClient client.BucketClient) Downloadable {
```

Comment:
FYI for other reviewers: I suggested this to keep the changeset minimal for the start. When we moved more stuff, we can improve this file/package more easily.